Do not bin relation fields even if they have fingerprints

pawit-metabase opened the pull request:
Fixes #17768

pawit-metabase committed:
Do not bin relation fields even if they have fingerprints

Fixes #17768

camsaul commented on test/metabase/api/table_test.clj:
```diff
@@ -664,6 +664,13 @@
                    (extract-dimension-options response "price"))))))
 
+      (testing "Relationship fields should not have binning options"
+        (let [fingerprint (db/select-one-field :fingerprint Field {:id (mt/id :venues :price)})]
```
This query doesn't do what you think it's doing. You need to write either

```clj
(db/select-one-field :fingerprint Field :id (mt/id :venues :price))
```

or

```clj
(db/select-one-field :fingerprint Field {:where [:=:id (mt/id :venues :price)]})
```

Maps passed to Toucan function are just treated as raw HoneySQL. `:id` in a HoneySQL map just gets ignored

camsaul replied:
Can you add an assertion like  

```clj
(is (seq (extra-dimension-options response "id")))
```

_before_ adding the fingerprint information? So the whole test looks like

1. before changing anything
   1. test that dimension options get returned
1. add Fingerprint info
   1. test that dimension options no longer get returned

otherwise for all we know the change you made adding `:fingerprint` info had nothing to do with anything and the dimension options wouldn't have been returned either way

camsaul commented on test/metabase/api/table_test.clj:
```diff
@@ -664,6 +664,13 @@
+      (testing "Relationship fields should not have binning options"
+        (let [fingerprint (db/select-one-field :fingerprint Field {:id (mt/id :venues :price)})]
+          (mt/with-temp-vals-in-db Field (mt/id :venues :id) {:fingerprint fingerprint}
+           (let [response (mt/user-http-request :rasta :get 200 (format "table/%d/query_metadata" (mt/id :venues)))]
```
this should be indented two spaces

camsaul reviewed (changes requested):
A couple of small things:

1. Need to make a small tweak to the `db/select-one-field` call
2. Let's add an extra test to make sure `extra-dimension-options` were being returned in the first place and that the change we made causes them to no longer be returned (as opposed to them not being returned regardless of this change)
3. Minor formatting notes

pawit-metabase committed:
Fix test from review

pawit-metabase commented:
> Need to make a small tweak to the db/select-one-field call 

It's gone now after reworking the test.

However, I got it from the `Numeric fields without min/max values should not have binning options` test below. Let me fix it there.

> Let's add an extra test to make sure extra-dimension-options were being returned in the first place and that the change we made causes them to no longer be returned (as opposed to them not being returned regardless of this change)

With or without the fingerprint, the result should be the same - no dimension options. The bug was that dimension options were being returned if fingerprint information is present when it shouldn't be.

I've changed the test to use the `price` column instead. The "base" case is be provided by the `Number columns without a semantic type should use "num-bins"` test immediately above. Does that work?

> Minor formatting notes

Fixed
